tls/s2n_connection.c

```diff
@@ -231,6 +231,8 @@ struct s2n_connection *s2n_connection_new(s2n_mode mode)
      * is received, the stuffer will be resized according to the cookie length */
     GUARD_PTR(s2n_stuffer_growable_alloc(&conn->cookie_stuffer, 0));
 
+    conn->application_protocols_overridden = (struct s2n_blob){ 0 };
```

Comment:
You shouldn't actually need to do this. s2n_connection_new calls s2n_connection_wipe, which zero-inits the entire connection structure. The assumption is that we can't be trusted to remember to zero-init things :)

Reply:
Without this, I get some really weird state that doesn't make any sense in the tests.

Reply:
You might want to dig deeper on that-- this really shouldn't be necessary. We [zero the entire connection](https://github.com/awslabs/s2n/blob/main/tls/s2n_connection.c#L247-L250) when we [wipe the connection](https://github.com/awslabs/s2n/blob/main/tls/s2n_connection.c#L683), and we wipe the connection when we [create a new connection](https://github.com/awslabs/s2n/blob/main/tls/s2n_connection.c#L227).

Reply:
I've removed it and it still works. Must have been something else :/